libvirt/tests/src/sriov/sriov_nodedev.py

```diff
@@ -0,0 +1,125 @@
+    """
+    pci_list = ["0x%s" % x for x in re.split('[.:]', pci_id)]
+    addrs = dict(zip(['domain', 'bus', 'slot', 'function'], pci_list))
+    addrs.update({'type': 'pci'})
+    return pci_to_addr
+
+
+def add_hostdev_device(vm_name, pci):
+    """
+    Add hostdev device to VM
+
+    :param vm_name: Name of VM
+    :param pci_id: PCI ID of a device
+    """
+    vmxml = vm_xml.VMXML.new_from_inactive_dumpxml(vm_name)
+    hostdev_dev = libvirt.create_hostdev_xml(pci)
+    vmxml.add_device(hostdev_dev)
+    vmxml.sync()
+
+
+def run(test, params, env):
+    """
+    Nodedev related test.
+    """
+    def check_driver_from_xml(dev_name, driver_type='vfio-pci',
+                              status_error=False):
+        """
+        Check driver of a device from nodedev XML
+
+        :param dev_name: Name of a device(eg. pci_0000_05_00_1)
+        :param driver_type: Type of a device
+        :param status_error: Whether the driver should be same with 'driver_type'
+        :raise: TestFail if not match
+        """
+        dev_xml = nodedev_xml.NodedevXML.new_from_dumpxml(dev_name)
+        if status_error == (dev_xml.driver_name != driver_type):
+            test.fail("The driver %s should%s be '%s'."
+                      % (dev_xml.driver_name, ' not' if status_error else '',
+                         driver_type))
+
+    def nodedev_test(dev_name, status_error=False):
+        """
+        Execute virsh nodedev-* commands
+
+        :param dev_name: Name of a device(eg. pci_0000_05_00_1)
+        :param status_error: Whether the command should be failed
+        """
+        res = virsh.nodedev_reset(dev_name, debug=True)
+        libvirt.check_exit_status(res, status_error)
+        res = virsh.nodedev_detach(dev_name, debug=True)
+        libvirt.check_exit_status(res, status_error)
+        res = virsh.nodedev_reattach(dev_name, debug=True)
+        libvirt.check_exit_status(res, status_error)
+
+    def check_hostdev_device(vm_name):
+        """
+        Check hostdev device from VM's XML
+
+        :param vm_name: VM's name
+        :raise: TestFail if the hoodev device is not found
+        """
+        vmxml = vm_xml.VMXML.new_from_dumpxml(vm_name)
+        check_hostdev = vm_xml.VMXML.new_from_dumpxml(vm_name)\
+            .devices.by_device_tag('hostdev')
+        if not check_hostdev:
+            test.fail("The hostdev device does not exist: %s."
+                      % check_hostdev)
+
+    def test_pf():
+        dev_name = get_device_name(pf_pci)
+        check_driver_from_xml(dev_name)
+        nodedev_test(dev_name)
+        add_hostdev_device(vm_name, pf_pci)
+        vm.start()
+        check_hostdev_device(vm_name)
+        check_driver_from_xml(dev_name, status_error=True)
+        nodedev_test(dev_name, True)
+
+    test_case = params.get("test_case", "")
+    run_test = eval("test_%s" % test_case)
+
+    vm_name = params.get("main_vm", "avocado-vt-vm1")
+    vm = env.get_vm(params["main_vm"])
```

Comment:
```suggestion
    vm = env.get_vm(vm_name)
```
Seems you have get the value already?

Reply:
updated.